scripts/tool/synchronisation/SynchronizeAll.php

```diff
@@ -65,19 +70,38 @@ public function __invoke($params)
             $success = true;
         } catch (\Exception $e) {
             $report->add(Report::createFailure('An error has occurred : ' . $e->getMessage()));
+            $failureReason = $e->getMessage();
         } finally {
             if ($success === true) {
                 $event = new SyncFinishedEvent($params, $report);
             } else {
                 $report->add(Report::createFailure('An unexpected PHP error has occurred.'));
                 $event = new SyncFailedEvent($params, $report);
+                $event->setReason($failureReason);
             }
             $eventManager->trigger($event);
         }
 
         return $report;
     }
 
+    /**
+     * Check if all preconditions to perform synchronization are met.
+     *
+     * @param array $params
+     * @return bool
+     *
+     * @throws NotSupportedVmVersionException
+     */
+    private function checkSyncPreconditions(array $params)
+    {
+        $vmVersionChecker = $this->getServiceLocator()->get(VmVersionChecker::SERVICE_ID);
+
+        if (!$vmVersionChecker->isVmSupported($params['tao_version'])) {
+            throw new NotSupportedVmVersionException('Current version of TAO VM is not supported by the server.');
```

Comment:
Maybe it would make sense to mention the actual VM TAO version, so that central server administrator would know what is the version in the old VM.

Reply:
Fixed